bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,740 @@
+     *
+     * @param resources a list of incoming resource objects.
+     */
+    public void onResourcesEvent(List<Resource> resources) {
+        if (assetsLoaded) {
+            scheduler.submit(() -> {
+                logger.debug("onResourcesEvent() resource count {}", resources.size());
+                resources.forEach(resource -> onResource(resource));
+            });
+        }
+    }
+
+    /**
+     * Execute an HTTP PUT to send a Resource object to the server.
+     *
+     * @param resource the resource to put.
+     * @throws ApiException if a communication error occurred.
+     * @throws AssetNotLoadedException if one of the assets is not loaded.
+     */
+    public void putResource(Resource resource) throws ApiException, AssetNotLoadedException {
+        logger.debug("putResource() {}", resource);
+        checkAssetsLoaded();
+        getClip2Bridge().putResource(resource);
+    }
+
+    /**
+     * Register the application key with the hub. If the current application key is empty it will create a new one.
+     *
+     * @throws ApiException if a communication error occurred.
+     * @throws AssetNotLoadedException if one of the assets is not loaded.
+     * @throws HttpUnauthorizedException if the communication was OK but the registration failed anyway.
+     * @throws IllegalStateException if the configuration cannot be changed e.g. read only.
+     */
+    private void registerApplicationKey()
+            throws HttpUnauthorizedException, ApiException, AssetNotLoadedException, IllegalStateException {
+        logger.debug("registerApplicationKey()");
+        Clip2BridgeConfig config = getConfigAs(Clip2BridgeConfig.class);
+        String newApplicationKey = getClip2Bridge().registerApplicationKey(config.applicationKey);
+        Configuration configuration = editConfiguration();
+        configuration.put(Clip2BridgeConfig.APPLICATION_KEY, newApplicationKey);
+        updateConfiguration(configuration);
+    }
+
+    /**
+     * Register or unregister the discovery service.
+     *
+     * @param discoveryService new discoveryService, or null if un-registering.
+     */
+    public void registerDiscoveryService(@Nullable Clip2ThingDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    /**
+     * Update the bridge's online state and update its dependent things. Called when the connection goes online.
+     */
+    private void updateOnlineState() {
+        if (assetsLoaded && (thing.getStatus() != ThingStatus.ONLINE)) {
+            logger.debug("updateOnlineState()");
+            connectRetriesRemaining = RECONNECT_MAX_TRIES;
+            updateStatus(ThingStatus.ONLINE);
+            updateScenesAndThings(500);
+            Clip2ThingDiscoveryService discoveryService = this.discoveryService;
+            if (Objects.nonNull(discoveryService)) {
+                discoveryService.startScan(null);
```

Comment:
@andrewfg Here is where you initiate the scan, apparently whenever the bridge goes online?

Reply:
Yes. ??